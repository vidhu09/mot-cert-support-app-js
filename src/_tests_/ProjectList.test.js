import ProjectList from '../components/ProjectList';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import nock from 'nock';

nock('http://localhost')
   .persist()
   .get('/v1/project')
   .reply(200, [{
       "id" : "1",
       "name" : "Project 1",
       "description" : "This is the first project"
   },{
       "id" : "2",
       "name" : "Project 2",
       "description" : "This is the second project"
   }]);

describe('ProjectList', () => {

    it('renders the list of projects correctly', async () => {
        const {asFragment, findByText} = render(<MemoryRouter>
            <ProjectList />
        </MemoryRouter>);

await findByText('Project 1');

expect(asFragment()).toMatchSnapshot();

    });
 
 });