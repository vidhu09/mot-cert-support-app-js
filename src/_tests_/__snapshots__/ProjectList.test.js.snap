// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectList renders the list of projects correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row mt-5"
    >
      <div
        class="col-10"
      >
        <h1>
          Welcome to Timesheet manager
        </h1>
      </div>
      <div
        class="col-2"
      >
        <a
          class="float-end"
          href="/logout"
        >
          <svg
            fill="currentColor"
            height="56"
            viewBox="0 0 16 16"
            width="56"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M6 12.5a.5.5 0 0 0 .5.5h8a.5.5 0 0 0 .5-.5v-9a.5.5 0 0 0-.5-.5h-8a.5.5 0 0 0-.5.5v2a.5.5 0 0 1-1 0v-2A1.5 1.5 0 0 1 6.5 2h8A1.5 1.5 0 0 1 16 3.5v9a1.5 1.5 0 0 1-1.5 1.5h-8A1.5 1.5 0 0 1 5 12.5v-2a.5.5 0 0 1 1 0v2z"
              fill-rule="evenodd"
            />
            <path
              d="M.146 8.354a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L1.707 7.5H10.5a.5.5 0 0 1 0 1H1.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3z"
              fill-rule="evenodd"
            />
          </svg>
        </a>
        <a
          class="float-end"
          href="/profile"
        >
          <svg
            fill="currentColor"
            height="56"
            viewBox="0 0 16 16"
            width="56"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8 8a3 3 0 1 0 0-6 3 3 0 0 0 0 6Zm2-3a2 2 0 1 1-4 0 2 2 0 0 1 4 0Zm4 8c0 1-1 1-1 1H3s-1 0-1-1 1-4 6-4 6 3 6 4Zm-1-.004c-.001-.246-.154-.986-.832-1.664C11.516 10.68 10.289 10 8 10c-2.29 0-3.516.68-4.168 1.332-.678.678-.83 1.418-.832 1.664h10Z"
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      class="row mt-2"
    >
      <div
        class="col-12"
      >
        <div
          class="card"
        >
          <div
            class="card-body"
          >
            <h2
              class="card-title"
            >
              Projects
            </h2>
            <p>
              Select one of the projects to view your timesheet and add new entries.
            </p>
            <ul
              class="list-group"
            >
              <li
                class="list-group-item"
              >
                <a
                  href="/projects/1"
                >
                  Project 1
                </a>
              </li>
              <li
                class="list-group-item"
              >
                <a
                  href="/projects/2"
                >
                  Project 2
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
